homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -1,291 +1,368 @@
 from homeassistant.components.media_player.const import (
-    MEDIA_TYPE_MUSIC,
+    REPEAT_MODE_OFF,
+    SUPPORT_CLEAR_PLAYLIST,
     SUPPORT_NEXT_TRACK,
     SUPPORT_PAUSE,
     SUPPORT_PLAY,
     SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_REPEAT_SET,
+    SUPPORT_SELECT_SOUND_MODE,
     SUPPORT_SELECT_SOURCE,
+    SUPPORT_SHUFFLE_SET,
     SUPPORT_STOP,
     SUPPORT_TURN_OFF,
     SUPPORT_TURN_ON,
     SUPPORT_VOLUME_MUTE,
     SUPPORT_VOLUME_SET,
 )
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import (
     CONF_HOST,
-    CONF_PORT,
     STATE_IDLE,
-    STATE_ON,
+    STATE_OFF,
     STATE_PAUSED,
     STATE_PLAYING,
-    STATE_UNKNOWN,
 )
-import homeassistant.helpers.config_validation as cv
-import homeassistant.util.dt as dt_util
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import DiscoveryInfoType, HomeAssistantType
+
+from . import MusicCastDataUpdateCoordinator, MusicCastDeviceEntity
+from .const import (
+    DEFAULT_ZONE,
+    DOMAIN,
+    HA_REPEAT_MODE_TO_MC_MAPPING,
+    MC_REPEAT_MODE_TO_HA_MAPPING,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
-SUPPORTED_FEATURES = (
-    SUPPORT_PLAY
-    | SUPPORT_PAUSE
-    | SUPPORT_STOP
-    | SUPPORT_PREVIOUS_TRACK
-    | SUPPORT_NEXT_TRACK
-    | SUPPORT_TURN_ON
-    | SUPPORT_TURN_OFF
+MUSIC_PLAYER_SUPPORT = (
+    SUPPORT_PAUSE
     | SUPPORT_VOLUME_SET
     | SUPPORT_VOLUME_MUTE
+    | SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+    | SUPPORT_CLEAR_PLAYLIST
+    | SUPPORT_PLAY
+    | SUPPORT_SHUFFLE_SET
+    | SUPPORT_REPEAT_SET
+    | SUPPORT_PREVIOUS_TRACK
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_SELECT_SOUND_MODE
     | SUPPORT_SELECT_SOURCE
+    | SUPPORT_STOP
 )
 
-KNOWN_HOSTS_KEY = "data_yamaha_musiccast"
-INTERVAL_SECONDS = "interval_seconds"
 
-DEFAULT_PORT = 5005
-DEFAULT_INTERVAL = 480
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config,
+    async_add_devices: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Import legacy configurations."""
+
+    if config.get(CONF_HOST, "") not in [
+        x.data[CONF_HOST] for x in hass.config_entries.async_entries(DOMAIN)
+    ] and len(hass.config_entries.async_entries(DOMAIN)):
+        _LOGGER.error(
+            "Configuration in configuration.yaml is not supported anymore. "
+            "Please add this device using the config flow: %s",
+            config.get(CONF_HOST, ""),
+        )
+    else:
+        _LOGGER.warning(
+            "Configuration in configuration.yaml is deprecated. Use the config flow instead"
+        )
+
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+            )
+        )
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We should keep the platform schema until we invalidate config YAML completely for the integration.